workers/unity/Assets/Playground/Scripts/Workers/UnityGameLogic.cs

```diff
@@ -11,11 +11,18 @@ public class UnityGameLogic : WorkerBase
 
         public override string GetWorkerType => WorkerType;
 
-        public UnityGameLogic(string workerId, Vector3 origin) : base(workerId, origin)
+        public UnityGameLogic(string workerId, Vector3 origin) : base(workerId, origin, new ForwardingDispatcher())
         {
             PlayerLifecycleConfig.CreatePlayerEntityTemplate = PlayerTemplate.CreatePlayerEntityTemplate;
         }
 
+        public override void Connect(ConnectionConfig config)
+        {
+            base.Connect(config);
+
+            ((ForwardingDispatcher) View.LogDispatcher).SetConnection(Connection);
```

Comment:
Why not initialize a field of the type `ForwardingDispatcher`, pass that to `base`, and use that here to avoid a pretty dodgy-looking cast?

Reply:
I don't think that is possible. The ForwardingDispatcher would have to be a static field for me to be able to pass it to `base` on construction. I guess I could it would be possible to create a method in base called SetDispatcher() but this would leave the constructor in an uninitialized state without setting the dispatcher.